src/modules/feature-modules/admin/pages/service-users/service-user-inactivate-role.component.ts

```diff
@@ -0,0 +1,103 @@
+import { getInactivateRoleUserRules } from './../../services/users-validation-rules.service';
+import { Component, OnInit } from '@angular/core';
+import { ActivatedRoute } from '@angular/router';
+
+import { CoreComponent } from '@app/base';
+import { RoutingHelper } from '@app/base/helpers';
+
+import { ServiceUsersService } from '../../services/service-users.service';
+import { UsersValidationRulesService } from '../../services/users-validation-rules.service';
+import { forkJoin } from 'rxjs/internal/observable/forkJoin';
+
+
+@Component({
+  selector: 'app-admin-pages-service-users-service-user-inactivate-role',
+  templateUrl: './service-user-inactivate-role.component.html'
+})
+export class PageServiceUserInactivateRoleComponent extends CoreComponent implements OnInit {
+
+  user: {
+    id: string,
+    name: string,
+    role: {
+      id: string;
+      description: string;
+    },
+   } = { id: '', name: '', role: { id: '', description: '' } };
+
+  pageStep: 'RULES' | 'INACTIVATE_ROLE' = 'RULES';
+
+  rulesList: getInactivateRoleUserRules['validations'] = [];
+
+
+  constructor(
+    private activatedRoute: ActivatedRoute,
+    private serviceUsersService: ServiceUsersService,
+    private usersValidationRulesService: UsersValidationRulesService
+  ) {
+
+    super();
+
+    this.user = {
+      ...this.user,
```

Comment:
I think there's no need to have this here since you assign values for each available property bellow.

Reply:
Thanks! I have changed this object a feel times, and turns out I don't need to make this anymore